cfme/tests/cloud_infra_common/test_provisioning.py

```diff
@@ -240,22 +231,21 @@ def test_provision_approval(appliance, provider, vm_name, smtp_test, request,
                                                                  provider).cleanup_on_provider()
                      for name in vm_names]
         )
-    subject = VersionPicker({
-        LOWEST: "your virtual machine configuration was approved",
-        "5.10": "your {} request was approved".format(vm_type)
-    }).pick()
+
     try:
-        wait_for(
-            lambda:
-            len(filter(
-                lambda mail:
-                normalize_text(subject) in normalize_text(mail["subject"]),
-                smtp_test.get_emails())) == 1,
-            num_sec=120, delay=5)
+        wait_for(lambda: len(smtp_test.get_emails()) >= 2,
+                 num_sec=90,
+                 delay=3)
     except TimedOutError:
-        subjects = ",".join([normalize_text(m["subject"]) for m in smtp_test.get_emails()])
-        logger.error("expected: %s, got emails: %s", subject, subjects)
-        raise
+        pytest.fail('Did not receive at least 1 emails from provisioning request, received: {}'
+                    .format(smtp_test.get_emails()))
+    received_subjects = [normalize_text(m["subject"]) for m in smtp_test.get_emails()]
```

Comment:
I'm not the biggest fan of my own naming/pattern here, if anyone can suggest better names or a more clear way to format this I'm all for it.  Gets a little messy with applying `normalize_text` on everything, and having to iterate in multiple layers (checking for 2 strings, checking for partial match of each string in a list of received strings).